sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/implementation/EncryptionProcessor.java

```diff
@@ -257,28 +257,44 @@ public EncryptionSettings getEncryptionSettings() {
             List<Mono<Void>> encryptionMonoList = new ArrayList<>();
             for (ClientEncryptionIncludedPath includedPath : this.clientEncryptionPolicy.getIncludedPaths()) {
                 String propertyName = includedPath.getPath().substring(1);
-                JsonNode propertyValueHolder = itemJObj.get(propertyName);
 
-                // Even null in the JSON is a JToken with Type Null, this null check is just a sanity check
-                if (propertyValueHolder != null && !propertyValueHolder.isNull()) {
-                    Mono<Void> voidMono = this.encryptionSettings.getEncryptionSettingForPropertyAsync(propertyName,
-                        this).flatMap(settings -> {
-                        try {
-                            encryptAndSerializeProperty(settings, itemJObj, propertyValueHolder, propertyName);
-                        } catch (MicrosoftDataEncryptionException ex) {
-                            return Mono.error(ex);
-                        }
-                        return Mono.empty();
-                    });
-                    encryptionMonoList.add(voidMono);
+                if (itemJObj.isValueNode()) {
+                    if (patchPropertyPath.substring(1).equals(propertyName)) {
+                        return this.encryptionSettings.getEncryptionSettingForPropertyAsync(propertyName,
+                            this).flatMap(settings -> {
+                                try{
+                                    return Mono.just(EncryptionUtils.getSimpleObjectMapper().readTree(EncryptionUtils.getSimpleObjectMapper().writeValueAsString(encryptAndSerializeValue(settings,
```

Comment:
Check id Jackson give direct api for byte[] to JsonNode

Reply:
Could not find a simpler way of doing it.